Source/WebCore/rendering/GridBaselineAlignment.cpp

```diff
@@ -76,7 +77,24 @@ LayoutUnit GridBaselineAlignment::ascentForChild(const RenderBox& child, GridAxi
             return isFlippedWritingMode(m_blockFlow) ? child.size().width().toInt() + margin : margin;
         return child.size().height() + margin;
     }
-    return baseline + margin;
+
+    auto subgridOffset = [&] {
+        if (position == ItemPosition::LastBaseline || baselineAxis == GridAxis::GridRowAxis)
+            return 0_lu;
+
+        if (const auto* subgrid = dynamicDowncast<RenderGrid>(child.parent()); subgrid && subgrid->isSubgridRows()) {
+            auto isPlacedAtFirstTrackInSubgrid = [&] {
+                return !subgrid->gridSpanForChild(child, GridTrackSizingDirection::ForRows).startLine();
+            }();
+
+            if (!isPlacedAtFirstTrackInSubgrid || GridLayoutFunctions::isOrthogonalParent(*subgrid, *subgrid->parent()))
+                return 0_lu;
+            return baselineAxis == GridAxis::GridColumnAxis ? subgrid->marginAndBorderAndPaddingBefore() : subgrid->marginAndBorderAndPaddingStart();
```

Comment:
Actually noticed that the else part of the ternary is dead code because of the if at the top. Removed